common/djangoapps/microsite_configuration/__init__.py

```diff
@@ -1 +1,19 @@
+from django.conf import settings as base_settings
+
+from microsite_configuration import microsite
 from .templatetags.microsite import page_title_breadcrumbs
+
+
+class MicrositeAwareSettings():
+    """
+    This class is a handy utility to make a call to the settings
+    completely microsite aware by replacing the:
+    from django.conf import settings
+    with:
+    from microsite_configuration import settings
+    """
+
+    def __getattr__(self, name):
+        return microsite.get_value(name, base_settings.__getattr__(name))
+
+settings = MicrositeAwareSettings()
```

Comment:
Thinking a bit more, this approach does expose a mistake I've made in the beginning of Microsites. One of the most common config parameters to override is PLATFORM_NAME, but as a config override I did not preserve the capitalization, so it appears in microsite configuration as "platform_name". I think think gap is significant as your __getattr__ wouldn't work in that case, because the attribute/key names don't match exactly.

Let me think about the significance of this. It might be best to bite the bullet and rectify this, but I'd like to think through what that might mean from a practical point of view.